### DISCUSSION TITLE
How to dynamically add more handles to a node?

### DISCUSSION THREAD

**Discussion by localhosted** [xyflow/xyflow#2950] [Q&A] [answered]
I'm trying to add new handles as the user keeps adding edges, but the edges won't connect properly to the handle, just group toghether at the first handle instead:

![image](https://user-images.githubusercontent.com/8821661/227081133-366e4675-b6e0-48bf-aacd-3e964ca978f5.png)

https://codesandbox.io/s/react-flow-varied-handles-forked-9b6ce8

**Comment by moklick** [comment, marked as answer]
If you are dynamically add handles you need to use [`useUpdateNodeInternals`](https://reactflow.dev/docs/api/hooks/use-update-node-internals/) . It re-measures the positions of the handles and writes that information back to the nodes data. Example: https://github.com/wbkd/react-flow/blob/main/examples/vite-app/src/examples/UseUpdateNodeInternals/CustomNode.tsx#L25-L26

**Reply by Suhailbsth** [reply]
Is useUpdateInternal automatically arranges handle position if the handles incremented dynamically?

**Comment by localhosted** [comment]
Updated the codesandbox example and it works pretty well
